### PR TITLE
one more fix related to typed_span conversion

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: src/storage_utils.cpp [hunk 1]
```diff
@@ -174,7 +174,8 @@ namespace libtorrent { namespace aux {
 			if (ec) return -1;
 
 			// advance our position in the iovec array and the file offset.
-			current_buf = advance_bufs(current_buf, bytes_transferred);
```
Comment (arvidn) [inline src/storage_utils.cpp]: would you mind making ``advance_bufs`` just use ``typed_span<iovec_t, int>`` instead? I believe it's only used for that type anyway

Comment (aldenml) [inline src/storage_utils.cpp]: yes, I saw that option too, didn't do it because it involve more code, let me do that

**Comment by aldenml** [comment]
I think it looks better now, let me know

**Comment by arvidn** [comment]
lgtm

**Pull request merged**